src/EnergyPlus/HeatBalanceManager.cc

```diff
@@ -339,6 +340,13 @@ namespace HeatBalanceManager {
 		// Get the heat balance input at the beginning of the simulation only
 		if ( ManageHeatBalanceGetInputFlag ) {
 			GetHeatBalanceInput(); // Obtains heat balance related parameters from input file
+
+			// Surface octree setup
+			//  The surface octree holds live references to surfaces so it must be updated
+			//   if in the future surfaces are altered after this point
+			surfaceOctree.init( DataSurfaces::Surface ); // Set up surface octree
```

Comment:
And does octree apply for the DELight method?  Probably not?  If we use a shared octree, then it seems the test goes away and you would initialize any time the surface count threshold is exceeded since shading always happens.  And if it's just for reflection, that has a simple flag to check, CalcSolRefl.